File: google_search_cli.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "os/exec"
    "strings"
)

func main() {
    reader := bufio.NewReader(os.Stdin)
    fmt.Print("Enter your search query: ")
    query, _ := reader.ReadString('\n')
    query = strings.TrimSpace(query)

    url := "https://www.google.com/search?q=" + query

    var command *exec.Cmd
    if os.Getenv("OS") == "Windows_NT" {
        command = exec.Command("cmd", "/c", "start", url)
    } else if os.Getenv("OSTYPE") == "darwin" {
        command = exec.Command("open", url)
    } else {
        command = exec.Command("xdg-open", url)
    }

    err := command.Start()
    if err != nil {
        fmt.Println("Failed to open browser:", err)
    } else {
        fmt.Println("Searching for:", query)
    }
}

File: google_search_cli.hs
import System.Process (callCommand)
import Network.URI (escapeURIString, isUnreserved)

main :: IO ()
main = do
    putStrLn "Enter your search query: "
    query <- getLine

    let url = "https://www.google.com/search?q=" ++ escapeURIString isUnreserved query
    let command = case os of
                    "linux"   -> "xdg-open \"" ++ url ++ "\""
                    "windows" -> "start \"" ++ url ++ "\""
                    "darwin"  -> "open \"" ++ url ++ "\""
                    _         -> error "Unsupported OS"

    callCommand command
    putStrLn $ "Searching for: " ++ query

os :: String
os = case System.Info.os of
        "linux"   -> "linux"
        "mingw32" -> "windows"
        "darwin"  -> "darwin"
        _         -> error "Unsupported OS"

